src/scripts/events/EventList.js

```diff
@@ -1 +1,70 @@
-import {getEvent, useEvent} from "./EventProvider.js"
\ No newline at end of file
+import {getEvents, useEvents} from "./EventProvider.js"
+import {eventHTML} from "./EventCard.js"
+import {useFriendsByUserId, getFriends, useFriends} from "../friends/FriendProvider.js"
+
+const contentTarget=document.querySelector(".event-container")
+
+export const eventList=()=>{
+    //get events then get friends
+    getEvents()
+    .then(getFriends)
+    .then(()=>{
+        //assign friends to the friendsArray
+        const friendsArray=useFriendsByUserId(sessionStorage.getItem("activeUser"))
+        //assign events to eventsArray
+        let eventArray=useEvents()
+        /*convert the date string in the event object into a date object.
+        Then assign the date object to the date key of the event object*/
+        if(eventArray.length!==0){
+            eventArray=eventArray.map(eventObj=>{
+            const dateObj=new Date(eventObj.date)
+            eventObj.date=dateObj
+            return eventObj
+        })
+        /*sorting the eventObject based on date then reverse it*/
+        const eventsSortedByDate=eventArray.sort((firstObj, secondObj)=>{
+            return secondObj.date-firstObj.date
+        }).reverse()
+        let correctEvents=[]
+        if(friendsArray.length===0){
+            correctEvents=eventsSortedByDate.filter(eventObj=>eventObj.userId===parseInt(sessionStorage.getItem("activeUser")))
```

Comment:
This condition isn't necessary, since the else condition still works if the friends array is empty. What you have works, but If you want to remove this, and just use the code that you have in the else condition, it would be easier to read. I don't see this as necessary, so I'll approve either way. 😃